### PR TITLE
fix: statsd hocon schema & add test suit

### PR DESCRIPTION
**Pull request by DDDHuang**
Fixes <issue-number>

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/emqx/emqx/blob/master/CONTRIBUTING.md.

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] In case of non-backward compatible changes, reviewer should check this item as a write-off, and add details in **Backward Compatibility** section

## More information

### CODE REVIEW

**Review by terry-xiaoyu — DISMISSED** [review]

File: apps/emqx_statsd/src/emqx_statsd_schema.erl [hunk 1]
```diff
@@ -17,22 +17,22 @@ fields("emqx_statsd") ->
     , {sample_time_interval, fun duration_s/1}
     , {flush_time_interval,  fun duration_s/1}].
 
-server(type) -> string();
+server(type) -> emqx_schema:ip_port();
```
Comment (zmstone) [inline apps/emqx_statsd/src/emqx_statsd_schema.erl:20]: this will fail if it's FQDN or a hostname configured

Comment (zmstone) [reply apps/emqx_statsd/src/emqx_statsd_schema.erl:20]: maybe we need a new type `emqx_schema:host_port()`

File: apps/emqx_statsd/test/emqx_statsd_SUITE.erl [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+t_statsd(_) ->
+    {ok, Socket} = gen_udp:open(8125),
+    receive
+        {udp, _Socket, _Host, _Port, Bin} ->
+            ?assert(length(Bin) > 50)
+    after
+        11*1000 ->
+            ?assert(true, failed)
+    end,
+    gen_udp:close(Socket).
```
Comment (zmstone) [inline apps/emqx_statsd/test/emqx_statsd_SUITE.erl:28]: please ensure line-break at EOF

Comment (DDDHuang) [reply apps/emqx_statsd/test/emqx_statsd_SUITE.erl]: fixed

**Review by zmstone — APPROVED** [review]

**Review by turtleDeng — APPROVED** [review]

**Pull request merged**